source/material_model/visco_plastic.cc

```diff
@@ -351,6 +351,7 @@ namespace aspect
 
               const SymmetricTensor<2,dim> strain_rate_difference = in.strain_rate[i]
                                                                     + std::max(std::fabs(in.strain_rate[i][strain_rate_indices]), min_strain_rate)
+                                                                    * (component > dim-1 ? 0.5 : 1 )
                                                                     * finite_difference_accuracy
                                                                     * Utilities::nth_basis_for_symmetric_tensors<dim>(component);
```

Comment:
```suggestion
               // components that are not on the diagonal are multiplied by 0.5, because the symmetric tensor
               // is modified by 0.5 in both symmetric directions (xy/yx) simultaneously and we compute the combined
               // derivative
              const SymmetricTensor<2,dim> strain_rate_difference = in.strain_rate[i]
                                                                    + std::max(std::fabs(in.strain_rate[i][strain_rate_indices]), min_strain_rate)
                                                                    * (component > dim-1 ? 0.5 : 1 )
                                                                    * finite_difference_accuracy
                                                                    * Utilities::nth_basis_for_symmetric_tensors<dim>(component);
```